### PR TITLE
edid-generator: 2018-03-15 -> 2023-11-15

### PR DESCRIPTION
**Pull request by nazarewk**
## Description of changes

result of:
- suprise merge of https://github.com/akatrevorjay/edid-generator/pull/28
- temporarily builds off my fork until https://github.com/akatrevorjay/edid-generator/pull/29 is merged

also:
- fixes and gives actually working example of `overrideAttrs`
- ensures it works with `hardware.firmware` option
- split into phases
- adds `checkPhase`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/edid-generator/default.nix [hunk 1]
```diff
@@ -5,43 +5,65 @@
...
   pname = "edid-generator";
-  version = "unstable-2018-03-15";
+  version = "master-2023-11-15";
+
+  # so `hardware.firmware` doesn't compress it
+  compressFirmware = false;
 
   src = fetchFromGitHub {
-    owner = "akatrevorjay";
+    # switch back after https://github.com/akatrevorjay/edid-generator/pull/29 is merged
```
Comment (flokli) [inline pkgs/tools/misc/edid-generator/default.nix:28]: Please do a fetchpatch with the GitHub patch URL, and add a comment linking to your PR I'm the line before the patch.

**Review by flokli — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/edid-generator/default.nix (hunk 1 above)
Comment (nazarewk) [reply pkgs/tools/misc/edid-generator/default.nix]: great idea

File: pkgs/tools/misc/edid-generator/default.nix [hunk 2]
```diff
@@ -1,47 +1,75 @@
...
     repo = "edid-generator";
-    rev = "31a6f80784d289d2faa8c4ca4788409c83b3ea14";
-    sha256 = "0j6wqzx5frca8b5i6812vvr5iwk7440fka70bmqn00k0vfhsc2x3";
+    rev = "9430c121e0b31d8d60799379f73722f08f2e62a1";
+    sha256 = "sha256-CoEcAl5680fCOF2XxxGSiBs6AqdsbBbaM+ENAC2lOU8=";
   };
+  patches = [
+    (fetchpatch {
+      # https://github.com/akatrevorjay/edid-generator/pull/29
+      url = "https://github.com/nazarewk/edid-generator/compare/9430c121e0b31d8d60799379f73722f08f2e62a1...cc9572bb94cdde0ff6e930bfd9354a3c8badc168.patch";
```
Comment (flokli) [inline pkgs/tools/misc/edid-generator/default.nix:37]:
```suggestion
      url = "https://github.com/akatrevorjay/edid-generator/pull/29/commits/cc9572bb94cdde0ff6e930bfd9354a3c8badc168.patch";
```

**Comment by nazarewk** [comment]
@flokli the patch got actually merged, so it's a lot simpler

File: pkgs/tools/misc/edid-generator/default.nix [hunk 3]
```diff
@@ -1,47 +1,68 @@
 { lib
 , stdenv
 , fetchFromGitHub
+, fetchpatch
```
Comment (flokli) [inline pkgs/tools/misc/edid-generator/default.nix:4]: now you can drop this again :-)

**Review by flokli — APPROVED** [review]

**Comment by flokli** [comment]
I'm too tired waiting for ofborg, merging this.

**Pull request merged**